app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -174,51 +153,66 @@
         rootScopeWatches = [];
       };
 
-      var hideIfNoProject = function(projectName) {
-        if(!projectName) {
-          drawer.drawerHidden = true;
+      var deregisterEventsWatch = function() {
+        if(eventsWatcher) {
+          DataService.unwatch(eventsWatcher);
         }
       };
 
-      var render = function() {
-        $rootScope.$evalAsync(function () {
-          countUnreadNotificationsForAllGroups();
-          // NOTE: we are currently only showing one project in the drawer at a
-          // time. If we go back to multiple projects, we can eliminate the filter here
-          // and just pass the whole array as notificationGroups.
-          // if we do, we will have to handle group.open to keep track of what the
-          // user is viewing at the time & indicate to the user that the non-active
-          // project is "asleep"/not being watched.
-          drawer.notificationGroups = _.filter(notificationGroups, function(group) {
-            return group.project.metadata.name === $routeParams.project;
-          });
-        });
+      var deregisterNotificationListener = function() {
+        notificationListener && notificationListener();
+        notificationListener = null;
       };
 
-      // TODO: follow-on PR to decide which of these events to toast,
-      // via config in constants.js
       var eventWatchCallback = function(eventData) {
-        eventsByNameData = eventData.by('metadata.name');
-        eventsMap = formatAndFilterEvents(eventsByNameData);
-        // TODO: Update to an intermediate map, so that we can then combine both
-        // events + notifications into the final notificationGroups output
-        notificationGroups = sortNotificationGroups(eventsMap);
+        eventsMap[$routeParams.project] = formatAndFilterEvents(eventData.by('metadata.name'));
         render();
       };
 
-      // TODO: Follow-on PR to update & add the internal notifications to the
-      // var notificationWatchCallback = function(event, notification) {
-      //   // will need to add .event = {} and immitate structure
-      //   if(!notification.lastTimestamp) {
-      //     // creates a timestamp that matches event format: 2017-08-09T19:55:35Z
-      //     notification.lastTimestamp = moment.parseZone(new Date()).utc().format();
-      //   }
-      //   clientGeneratedNotifications.push(notification);
-      // };
+      var notificationWatchCallback = function(event, notification) {
+        var uid = notification.id || _.uniqueId('notification-');
+        var project = notification.namespace || $routeParams.project;
+        notificationsMap[project] = notificationsMap[project] || {};
+        notificationsMap[project][uid] = {
+          actions: null,
+          unread: true,
+          uid: uid,
+          // emulating an API event to simplify the template
+          event: {
```

Comment:
I'd rather not add a dummy event. I'd rather copy necessary properties like `uid` and `timestamp` from the event to top-level properties, and use those in the view.

Reply:
ok ill update